[{'name': 'data_science_project_logger', 'msg': 'Custom logger is working fine!', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 10, 'funcName': '<module>', 'created': 1758702956.65264, 'msecs': 652.0, 'relativeCreated': 0.0, 'thread': 22492, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 34544, 'message': 'Custom logger is working fine!'}sctime)s: INFO: main: Custom logger is working fine!]
[{'name': 'data_science_project_logger', 'msg': 'Custom logger is working fine!', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 10, 'funcName': '<module>', 'created': 1758787165.4461162, 'msecs': 446.0, 'relativeCreated': 7.834196090698242, 'thread': 35280, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 29564, 'message': 'Custom logger is working fine!'}sctime)s: INFO: main: Custom logger is working fine!]
[{'name': 'data_science_project_logger', 'msg': '>>>>> stage Data Ingestion Stage started <<<<<', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 14, 'funcName': '<module>', 'created': 1758787259.2313285, 'msecs': 231.0, 'relativeCreated': 5971.374273300171, 'thread': 13168, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 35588, 'message': '>>>>> stage Data Ingestion Stage started <<<<<'}sctime)s: INFO: main: >>>>> stage Data Ingestion Stage started <<<<<]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: config\\config.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758787259.2313285, 'msecs': 231.0, 'relativeCreated': 5971.374273300171, 'thread': 13168, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 35588, 'message': 'yaml file: config\\config.yaml loaded successfully'}sctime)s: INFO: common: yaml file: config\config.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: params.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758787259.2467997, 'msecs': 246.0, 'relativeCreated': 5986.84549331665, 'thread': 13168, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 35588, 'message': 'yaml file: params.yaml loaded successfully'}sctime)s: INFO: common: yaml file: params.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': BoxValueError('yaml file is empty'), 'args': (), 'levelname': 'ERROR', 'levelno': 40, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': (<class 'box.exceptions.BoxValueError'>, BoxValueError('yaml file is empty'), <traceback object at 0x000002A9AD744180>), 'exc_text': None, 'stack_info': None, 'lineno': 19, 'funcName': '<module>', 'created': 1758787259.2467997, 'msecs': 246.0, 'relativeCreated': 5986.84549331665, 'thread': 13168, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 35588, 'message': 'yaml file is empty'}sctime)s: ERROR: main: yaml file is empty]
Traceback (most recent call last):
  File "D:\Data Science\src\DataScienceProject\utils\common.py", line 18, in read_yaml
    return ConfigBox(content)
           ^^^^^^^^^^^^^^^^^^
  File "box\\box.py", line 295, in box.box.Box.__init__
box.exceptions.BoxValueError: First argument must be mapping or iterable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:\Data Science\main.py", line 16, in <module>
    obj.initiate_data_ingestion()
  File "D:\Data Science\src\DataScienceProject\pipeline\data_ingestion_pipeline.py", line 12, in initiate_data_ingestion
    config=ConfigurationManager()
           ^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Data Science\src\DataScienceProject\config\configuration.py", line 10, in __init__
    self.params = read_yaml(params_filepath)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Data Science\.venv\Lib\site-packages\ensure\main.py", line 872, in __call__
    return_val = self.f(*args, **kwargs)
                 ^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Data Science\src\DataScienceProject\utils\common.py", line 20, in read_yaml
    raise BoxValueError("yaml file is empty")
box.exceptions.BoxValueError: yaml file is empty
[{'name': 'data_science_project_logger', 'msg': '>>>>> stage Data Ingestion Stage started <<<<<', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 14, 'funcName': '<module>', 'created': 1758787337.1517835, 'msecs': 151.0, 'relativeCreated': 257.4315071105957, 'thread': 30920, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 24112, 'message': '>>>>> stage Data Ingestion Stage started <<<<<'}sctime)s: INFO: main: >>>>> stage Data Ingestion Stage started <<<<<]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: config\\config.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758787337.1517835, 'msecs': 151.0, 'relativeCreated': 257.4315071105957, 'thread': 30920, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 24112, 'message': 'yaml file: config\\config.yaml loaded successfully'}sctime)s: INFO: common: yaml file: config\config.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: params.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758787337.153795, 'msecs': 153.0, 'relativeCreated': 259.4430446624756, 'thread': 30920, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 24112, 'message': 'yaml file: params.yaml loaded successfully'}sctime)s: INFO: common: yaml file: params.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: schema.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758787337.153795, 'msecs': 153.0, 'relativeCreated': 259.4430446624756, 'thread': 30920, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 24112, 'message': 'yaml file: schema.yaml loaded successfully'}sctime)s: INFO: common: yaml file: schema.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758787337.153795, 'msecs': 153.0, 'relativeCreated': 259.4430446624756, 'thread': 30920, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 24112, 'message': 'created directory at : artifacts'}sctime)s: INFO: common: created directory at : artifacts]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts/data_ingestion', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758787337.153795, 'msecs': 153.0, 'relativeCreated': 259.4430446624756, 'thread': 30920, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 24112, 'message': 'created directory at : artifacts/data_ingestion'}sctime)s: INFO: common: created directory at : artifacts/data_ingestion]
[{'name': 'data_science_project_logger', 'msg': FileNotFoundError(2, 'No such file or directory'), 'args': (), 'levelname': 'ERROR', 'levelno': 40, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': (<class 'FileNotFoundError'>, FileNotFoundError(2, 'No such file or directory'), <traceback object at 0x00000167AA1FA100>), 'exc_text': None, 'stack_info': None, 'lineno': 19, 'funcName': '<module>', 'created': 1758787338.175241, 'msecs': 175.0, 'relativeCreated': 1280.8890342712402, 'thread': 30920, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 24112, 'message': "[Errno 2] No such file or directory: 'artifcats/data_ingestion/data.zip'"}sctime)s: ERROR: main: [Errno 2] No such file or directory: 'artifcats/data_ingestion/data.zip']
Traceback (most recent call last):
  File "D:\Data Science\main.py", line 16, in <module>
    obj.initiate_data_ingestion()
  File "D:\Data Science\src\DataScienceProject\pipeline\data_ingestion_pipeline.py", line 15, in initiate_data_ingestion
    data_ingestion.download_file()
  File "D:\Data Science\src\DataScienceProject\components\data_ingestion.py", line 13, in download_file
    filename, headers= request.urlretrieve(
                       ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\AppData\Roaming\uv\python\cpython-3.11.12-windows-x86_64-none\Lib\urllib\request.py", line 251, in urlretrieve
    tfp = open(filename, 'wb')
          ^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: 'artifcats/data_ingestion/data.zip'
[{'name': 'data_science_project_logger', 'msg': '>>>>> stage Data Ingestion Stage started <<<<<', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 14, 'funcName': '<module>', 'created': 1758787532.280677, 'msecs': 280.0, 'relativeCreated': 320.7056522369385, 'thread': 37364, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 30088, 'message': '>>>>> stage Data Ingestion Stage started <<<<<'}sctime)s: INFO: main: >>>>> stage Data Ingestion Stage started <<<<<]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: config\\config.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758787532.2882183, 'msecs': 288.0, 'relativeCreated': 328.2468318939209, 'thread': 37364, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 30088, 'message': 'yaml file: config\\config.yaml loaded successfully'}sctime)s: INFO: common: yaml file: config\config.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: params.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758787532.2882183, 'msecs': 288.0, 'relativeCreated': 328.2468318939209, 'thread': 37364, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 30088, 'message': 'yaml file: params.yaml loaded successfully'}sctime)s: INFO: common: yaml file: params.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: schema.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758787532.2882183, 'msecs': 288.0, 'relativeCreated': 328.2468318939209, 'thread': 37364, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 30088, 'message': 'yaml file: schema.yaml loaded successfully'}sctime)s: INFO: common: yaml file: schema.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758787532.2882183, 'msecs': 288.0, 'relativeCreated': 328.2468318939209, 'thread': 37364, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 30088, 'message': 'created directory at : artifacts'}sctime)s: INFO: common: created directory at : artifacts]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts/data_ingestion', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758787532.2882183, 'msecs': 288.0, 'relativeCreated': 328.2468318939209, 'thread': 37364, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 30088, 'message': 'created directory at : artifacts/data_ingestion'}sctime)s: INFO: common: created directory at : artifacts/data_ingestion]
[{'name': 'data_science_project_logger', 'msg': FileNotFoundError(2, 'No such file or directory'), 'args': (), 'levelname': 'ERROR', 'levelno': 40, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': (<class 'FileNotFoundError'>, FileNotFoundError(2, 'No such file or directory'), <traceback object at 0x0000016AE361DCC0>), 'exc_text': None, 'stack_info': None, 'lineno': 19, 'funcName': '<module>', 'created': 1758787532.8843224, 'msecs': 884.0, 'relativeCreated': 924.3509769439697, 'thread': 37364, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 30088, 'message': "[Errno 2] No such file or directory: 'artifcats/data_ingestion/data.zip'"}sctime)s: ERROR: main: [Errno 2] No such file or directory: 'artifcats/data_ingestion/data.zip']
Traceback (most recent call last):
  File "D:\Data Science\main.py", line 16, in <module>
    obj.initiate_data_ingestion()
  File "D:\Data Science\src\DataScienceProject\pipeline\data_ingestion_pipeline.py", line 15, in initiate_data_ingestion
    data_ingestion.download_file()
  File "D:\Data Science\src\DataScienceProject\components\data_ingestion.py", line 13, in download_file
    filename, headers= request.urlretrieve(
                       ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\AppData\Roaming\uv\python\cpython-3.11.12-windows-x86_64-none\Lib\urllib\request.py", line 251, in urlretrieve
    tfp = open(filename, 'wb')
          ^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: 'artifcats/data_ingestion/data.zip'
[{'name': 'data_science_project_logger', 'msg': '>>>>> stage Data Ingestion Stage started <<<<<', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 14, 'funcName': '<module>', 'created': 1758787960.5344815, 'msecs': 534.0, 'relativeCreated': 260.71643829345703, 'thread': 26476, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 36780, 'message': '>>>>> stage Data Ingestion Stage started <<<<<'}sctime)s: INFO: main: >>>>> stage Data Ingestion Stage started <<<<<]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: config\\config.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758787960.5344815, 'msecs': 534.0, 'relativeCreated': 260.71643829345703, 'thread': 26476, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 36780, 'message': 'yaml file: config\\config.yaml loaded successfully'}sctime)s: INFO: common: yaml file: config\config.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: params.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758787960.5344815, 'msecs': 534.0, 'relativeCreated': 260.71643829345703, 'thread': 26476, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 36780, 'message': 'yaml file: params.yaml loaded successfully'}sctime)s: INFO: common: yaml file: params.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: schema.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758787960.5344815, 'msecs': 534.0, 'relativeCreated': 260.71643829345703, 'thread': 26476, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 36780, 'message': 'yaml file: schema.yaml loaded successfully'}sctime)s: INFO: common: yaml file: schema.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758787960.5344815, 'msecs': 534.0, 'relativeCreated': 260.71643829345703, 'thread': 26476, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 36780, 'message': 'created directory at : artifacts'}sctime)s: INFO: common: created directory at : artifacts]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts/data_ingestion', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758787960.5344815, 'msecs': 534.0, 'relativeCreated': 260.71643829345703, 'thread': 26476, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 36780, 'message': 'created directory at : artifacts/data_ingestion'}sctime)s: INFO: common: created directory at : artifacts/data_ingestion]
[{'name': 'data_science_project_logger', 'msg': 'file : artifacts/data_ingestion/data.zip downloaded successfully with following info: Connection: close\nContent-Length: 23329\nCache-Control: max-age=300\nContent-Security-Policy: default-src \'none\'; style-src \'unsafe-inline\'; sandbox\nContent-Type: application/zip\nETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"\nStrict-Transport-Security: max-age=31536000\nX-Content-Type-Options: nosniff\nX-Frame-Options: deny\nX-XSS-Protection: 1; mode=block\nX-GitHub-Request-Id: F166:246A48:32A99:619FF:68D4F709\nAccept-Ranges: bytes\nDate: Thu, 25 Sep 2025 08:12:41 GMT\nVia: 1.1 varnish\nX-Served-By: cache-maa10234-MAA\nX-Cache: HIT\nX-Cache-Hits: 0\nX-Timer: S1758787962.595613,VS0,VE310\nVary: Authorization,Accept-Encoding\nAccess-Control-Allow-Origin: *\nCross-Origin-Resource-Policy: cross-origin\nX-Fastly-Request-ID: ffdc48c36081ef07b20b420bfab4ea7e3a890dd2\nExpires: Thu, 25 Sep 2025 08:17:41 GMT\nSource-Age: 0\n\n', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\components\\data_ingestion.py', 'filename': 'data_ingestion.py', 'module': 'data_ingestion', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 18, 'funcName': 'download_file', 'created': 1758787961.464723, 'msecs': 464.0, 'relativeCreated': 1190.958023071289, 'thread': 26476, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 36780, 'message': 'file : artifacts/data_ingestion/data.zip downloaded successfully with following info: Connection: close\nContent-Length: 23329\nCache-Control: max-age=300\nContent-Security-Policy: default-src \'none\'; style-src \'unsafe-inline\'; sandbox\nContent-Type: application/zip\nETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"\nStrict-Transport-Security: max-age=31536000\nX-Content-Type-Options: nosniff\nX-Frame-Options: deny\nX-XSS-Protection: 1; mode=block\nX-GitHub-Request-Id: F166:246A48:32A99:619FF:68D4F709\nAccept-Ranges: bytes\nDate: Thu, 25 Sep 2025 08:12:41 GMT\nVia: 1.1 varnish\nX-Served-By: cache-maa10234-MAA\nX-Cache: HIT\nX-Cache-Hits: 0\nX-Timer: S1758787962.595613,VS0,VE310\nVary: Authorization,Accept-Encoding\nAccess-Control-Allow-Origin: *\nCross-Origin-Resource-Policy: cross-origin\nX-Fastly-Request-ID: ffdc48c36081ef07b20b420bfab4ea7e3a890dd2\nExpires: Thu, 25 Sep 2025 08:17:41 GMT\nSource-Age: 0\n\n'}sctime)s: INFO: data_ingestion: file : artifacts/data_ingestion/data.zip downloaded successfully with following info: Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: F166:246A48:32A99:619FF:68D4F709
Accept-Ranges: bytes
Date: Thu, 25 Sep 2025 08:12:41 GMT
Via: 1.1 varnish
X-Served-By: cache-maa10234-MAA
X-Cache: HIT
X-Cache-Hits: 0
X-Timer: S1758787962.595613,VS0,VE310
Vary: Authorization,Accept-Encoding
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: ffdc48c36081ef07b20b420bfab4ea7e3a890dd2
Expires: Thu, 25 Sep 2025 08:17:41 GMT
Source-Age: 0

]
[{'name': 'data_science_project_logger', 'msg': 'extracted all the files to : artifacts/data_ingestion', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\components\\data_ingestion.py', 'filename': 'data_ingestion.py', 'module': 'data_ingestion', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 27, 'funcName': 'extract_zip_file', 'created': 1758787961.4945073, 'msecs': 494.0, 'relativeCreated': 1220.7422256469727, 'thread': 26476, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 36780, 'message': 'extracted all the files to : artifacts/data_ingestion'}sctime)s: INFO: data_ingestion: extracted all the files to : artifacts/data_ingestion]
[{'name': 'data_science_project_logger', 'msg': '>>>>> stage Data Ingestion Stage completed <<<<<\n\nx==========x', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': '<module>', 'created': 1758787961.4948163, 'msecs': 494.0, 'relativeCreated': 1221.0512161254883, 'thread': 26476, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 36780, 'message': '>>>>> stage Data Ingestion Stage completed <<<<<\n\nx==========x'}sctime)s: INFO: main: >>>>> stage Data Ingestion Stage completed <<<<<

x==========x]
[{'name': 'data_science_project_logger', 'msg': '>>>>> stage Data Ingestion Stage started <<<<<', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 15, 'funcName': '<module>', 'created': 1758792948.6925485, 'msecs': 692.0, 'relativeCreated': 10585.471153259277, 'thread': 35292, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 24044, 'message': '>>>>> stage Data Ingestion Stage started <<<<<'}sctime)s: INFO: main: >>>>> stage Data Ingestion Stage started <<<<<]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: config\\config.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758792948.7242095, 'msecs': 724.0, 'relativeCreated': 10617.132186889648, 'thread': 35292, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 24044, 'message': 'yaml file: config\\config.yaml loaded successfully'}sctime)s: INFO: common: yaml file: config\config.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: params.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758792948.7262065, 'msecs': 726.0, 'relativeCreated': 10619.129180908203, 'thread': 35292, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 24044, 'message': 'yaml file: params.yaml loaded successfully'}sctime)s: INFO: common: yaml file: params.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: schema.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758792948.757661, 'msecs': 757.0, 'relativeCreated': 10650.583744049072, 'thread': 35292, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 24044, 'message': 'yaml file: schema.yaml loaded successfully'}sctime)s: INFO: common: yaml file: schema.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758792948.7596586, 'msecs': 759.0, 'relativeCreated': 10652.581214904785, 'thread': 35292, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 24044, 'message': 'created directory at : artifacts'}sctime)s: INFO: common: created directory at : artifacts]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts/data_ingestion', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758792948.7626576, 'msecs': 762.0, 'relativeCreated': 10655.580282211304, 'thread': 35292, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 24044, 'message': 'created directory at : artifacts/data_ingestion'}sctime)s: INFO: common: created directory at : artifacts/data_ingestion]
[{'name': 'data_science_project_logger', 'msg': 'FIle already exists', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\components\\data_ingestion.py', 'filename': 'data_ingestion.py', 'module': 'data_ingestion', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 20, 'funcName': 'download_file', 'created': 1758792948.763924, 'msecs': 763.0, 'relativeCreated': 10656.846523284912, 'thread': 35292, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 24044, 'message': 'FIle already exists'}sctime)s: INFO: data_ingestion: FIle already exists]
[{'name': 'data_science_project_logger', 'msg': 'extracted all the files to : artifacts/data_ingestion', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\components\\data_ingestion.py', 'filename': 'data_ingestion.py', 'module': 'data_ingestion', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 27, 'funcName': 'extract_zip_file', 'created': 1758792948.7732127, 'msecs': 773.0, 'relativeCreated': 10666.135311126709, 'thread': 35292, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 24044, 'message': 'extracted all the files to : artifacts/data_ingestion'}sctime)s: INFO: data_ingestion: extracted all the files to : artifacts/data_ingestion]
[{'name': 'data_science_project_logger', 'msg': '>>>>> stage Data Ingestion Stage completed <<<<<\n\nx==========x', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 18, 'funcName': '<module>', 'created': 1758792948.7742088, 'msecs': 774.0, 'relativeCreated': 10667.131423950195, 'thread': 35292, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 24044, 'message': '>>>>> stage Data Ingestion Stage completed <<<<<\n\nx==========x'}sctime)s: INFO: main: >>>>> stage Data Ingestion Stage completed <<<<<

x==========x]
[{'name': 'data_science_project_logger', 'msg': '>>>>> stage Data Validation Stage started <<<<<', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 25, 'funcName': '<module>', 'created': 1758792948.77521, 'msecs': 775.0, 'relativeCreated': 10668.132543563843, 'thread': 35292, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 24044, 'message': '>>>>> stage Data Validation Stage started <<<<<'}sctime)s: INFO: main: >>>>> stage Data Validation Stage started <<<<<]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: config\\config.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758792948.7782109, 'msecs': 778.0, 'relativeCreated': 10671.133518218994, 'thread': 35292, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 24044, 'message': 'yaml file: config\\config.yaml loaded successfully'}sctime)s: INFO: common: yaml file: config\config.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: params.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758792948.780603, 'msecs': 780.0, 'relativeCreated': 10673.52557182312, 'thread': 35292, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 24044, 'message': 'yaml file: params.yaml loaded successfully'}sctime)s: INFO: common: yaml file: params.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: schema.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758792948.78389, 'msecs': 783.0, 'relativeCreated': 10676.812648773193, 'thread': 35292, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 24044, 'message': 'yaml file: schema.yaml loaded successfully'}sctime)s: INFO: common: yaml file: schema.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758792948.78389, 'msecs': 783.0, 'relativeCreated': 10676.812648773193, 'thread': 35292, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 24044, 'message': 'created directory at : artifacts'}sctime)s: INFO: common: created directory at : artifacts]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts/data_validation', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758792948.7900114, 'msecs': 790.0, 'relativeCreated': 10682.934045791626, 'thread': 35292, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 24044, 'message': 'created directory at : artifacts/data_validation'}sctime)s: INFO: common: created directory at : artifacts/data_validation]
[{'name': 'data_science_project_logger', 'msg': PermissionError(13, 'Permission denied'), 'args': (), 'levelname': 'ERROR', 'levelno': 40, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': (<class 'PermissionError'>, PermissionError(13, 'Permission denied'), <traceback object at 0x0000029BD4736700>), 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': '<module>', 'created': 1758792948.793003, 'msecs': 793.0, 'relativeCreated': 10685.925722122192, 'thread': 35292, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 24044, 'message': "[Errno 13] Permission denied: 'artifacts/data_ingestion'"}sctime)s: ERROR: main: [Errno 13] Permission denied: 'artifacts/data_ingestion']
Traceback (most recent call last):
  File "D:\Data Science\main.py", line 27, in <module>
    obj.initiate_data_validation()
  File "D:\Data Science\src\DataScienceProject\pipeline\data_validation_pipeline.py", line 18, in initiate_data_validation
    validation_status=data_validation.validate_data_columns()
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Data Science\src\DataScienceProject\components\data_validation.py", line 29, in validate_data_columns
    raise e
  File "D:\Data Science\src\DataScienceProject\components\data_validation.py", line 13, in validate_data_columns
    data=pd.read_csv(self.config.unzip_data_dir)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Data Science\.venv\Lib\site-packages\pandas\io\parsers\readers.py", line 1026, in read_csv
    return _read(filepath_or_buffer, kwds)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Data Science\.venv\Lib\site-packages\pandas\io\parsers\readers.py", line 620, in _read
    parser = TextFileReader(filepath_or_buffer, **kwds)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Data Science\.venv\Lib\site-packages\pandas\io\parsers\readers.py", line 1620, in __init__
    self._engine = self._make_engine(f, self.engine)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Data Science\.venv\Lib\site-packages\pandas\io\parsers\readers.py", line 1880, in _make_engine
    self.handles = get_handle(
                   ^^^^^^^^^^^
  File "D:\Data Science\.venv\Lib\site-packages\pandas\io\common.py", line 873, in get_handle
    handle = open(
             ^^^^^
PermissionError: [Errno 13] Permission denied: 'artifacts/data_ingestion'
[{'name': 'data_science_project_logger', 'msg': '>>>>> stage Data Ingestion Stage started <<<<<', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 15, 'funcName': '<module>', 'created': 1758793159.753109, 'msecs': 753.0, 'relativeCreated': 610.074520111084, 'thread': 35196, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 33888, 'message': '>>>>> stage Data Ingestion Stage started <<<<<'}sctime)s: INFO: main: >>>>> stage Data Ingestion Stage started <<<<<]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: config\\config.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758793159.779722, 'msecs': 779.0, 'relativeCreated': 636.6875171661377, 'thread': 35196, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 33888, 'message': 'yaml file: config\\config.yaml loaded successfully'}sctime)s: INFO: common: yaml file: config\config.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: params.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758793159.779722, 'msecs': 779.0, 'relativeCreated': 636.6875171661377, 'thread': 35196, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 33888, 'message': 'yaml file: params.yaml loaded successfully'}sctime)s: INFO: common: yaml file: params.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: schema.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758793159.779722, 'msecs': 779.0, 'relativeCreated': 636.6875171661377, 'thread': 35196, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 33888, 'message': 'yaml file: schema.yaml loaded successfully'}sctime)s: INFO: common: yaml file: schema.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758793159.779722, 'msecs': 779.0, 'relativeCreated': 636.6875171661377, 'thread': 35196, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 33888, 'message': 'created directory at : artifacts'}sctime)s: INFO: common: created directory at : artifacts]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts/data_ingestion', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758793159.779722, 'msecs': 779.0, 'relativeCreated': 636.6875171661377, 'thread': 35196, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 33888, 'message': 'created directory at : artifacts/data_ingestion'}sctime)s: INFO: common: created directory at : artifacts/data_ingestion]
[{'name': 'data_science_project_logger', 'msg': 'FIle already exists', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\components\\data_ingestion.py', 'filename': 'data_ingestion.py', 'module': 'data_ingestion', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 20, 'funcName': 'download_file', 'created': 1758793159.779722, 'msecs': 779.0, 'relativeCreated': 636.6875171661377, 'thread': 35196, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 33888, 'message': 'FIle already exists'}sctime)s: INFO: data_ingestion: FIle already exists]
[{'name': 'data_science_project_logger', 'msg': 'extracted all the files to : artifacts/data_ingestion', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\components\\data_ingestion.py', 'filename': 'data_ingestion.py', 'module': 'data_ingestion', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 27, 'funcName': 'extract_zip_file', 'created': 1758793159.7951267, 'msecs': 795.0, 'relativeCreated': 652.0922183990479, 'thread': 35196, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 33888, 'message': 'extracted all the files to : artifacts/data_ingestion'}sctime)s: INFO: data_ingestion: extracted all the files to : artifacts/data_ingestion]
[{'name': 'data_science_project_logger', 'msg': '>>>>> stage Data Ingestion Stage completed <<<<<\n\nx==========x', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 18, 'funcName': '<module>', 'created': 1758793159.7951267, 'msecs': 795.0, 'relativeCreated': 652.0922183990479, 'thread': 35196, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 33888, 'message': '>>>>> stage Data Ingestion Stage completed <<<<<\n\nx==========x'}sctime)s: INFO: main: >>>>> stage Data Ingestion Stage completed <<<<<

x==========x]
[{'name': 'data_science_project_logger', 'msg': '>>>>> stage Data Validation Stage started <<<<<', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 25, 'funcName': '<module>', 'created': 1758793159.7961245, 'msecs': 796.0, 'relativeCreated': 653.0900001525879, 'thread': 35196, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 33888, 'message': '>>>>> stage Data Validation Stage started <<<<<'}sctime)s: INFO: main: >>>>> stage Data Validation Stage started <<<<<]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: config\\config.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758793159.7970605, 'msecs': 797.0, 'relativeCreated': 654.0260314941406, 'thread': 35196, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 33888, 'message': 'yaml file: config\\config.yaml loaded successfully'}sctime)s: INFO: common: yaml file: config\config.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: params.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758793159.8013253, 'msecs': 801.0, 'relativeCreated': 658.2908630371094, 'thread': 35196, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 33888, 'message': 'yaml file: params.yaml loaded successfully'}sctime)s: INFO: common: yaml file: params.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: schema.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758793159.8034089, 'msecs': 803.0, 'relativeCreated': 660.3744029998779, 'thread': 35196, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 33888, 'message': 'yaml file: schema.yaml loaded successfully'}sctime)s: INFO: common: yaml file: schema.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758793159.8034089, 'msecs': 803.0, 'relativeCreated': 660.3744029998779, 'thread': 35196, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 33888, 'message': 'created directory at : artifacts'}sctime)s: INFO: common: created directory at : artifacts]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts/data_validation', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758793159.804867, 'msecs': 804.0, 'relativeCreated': 661.8325710296631, 'thread': 35196, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 33888, 'message': 'created directory at : artifacts/data_validation'}sctime)s: INFO: common: created directory at : artifacts/data_validation]
[{'name': 'data_science_project_logger', 'msg': 'Data Validation is completed and all the columns are present in the data', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\pipeline\\data_validation_pipeline.py', 'filename': 'data_validation_pipeline.py', 'module': 'data_validation_pipeline', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 20, 'funcName': 'initiate_data_validation', 'created': 1758793159.8487465, 'msecs': 848.0, 'relativeCreated': 705.712080001831, 'thread': 35196, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 33888, 'message': 'Data Validation is completed and all the columns are present in the data'}sctime)s: INFO: data_validation_pipeline: Data Validation is completed and all the columns are present in the data]
[{'name': 'data_science_project_logger', 'msg': '>>>>> stage Data Validation Stage completed <<<<<\n\nx==========x', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 28, 'funcName': '<module>', 'created': 1758793159.8487465, 'msecs': 848.0, 'relativeCreated': 705.712080001831, 'thread': 35196, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 33888, 'message': '>>>>> stage Data Validation Stage completed <<<<<\n\nx==========x'}sctime)s: INFO: main: >>>>> stage Data Validation Stage completed <<<<<

x==========x]
[{'name': 'data_science_project_logger', 'msg': '>>>>> stage Data Ingestion Stage started <<<<<', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 16, 'funcName': '<module>', 'created': 1758795748.97428, 'msecs': 974.0, 'relativeCreated': 1364.6390438079834, 'thread': 18300, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 27344, 'message': '>>>>> stage Data Ingestion Stage started <<<<<'}sctime)s: INFO: main: >>>>> stage Data Ingestion Stage started <<<<<]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: config\\config.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758795748.9884515, 'msecs': 988.0, 'relativeCreated': 1378.8104057312012, 'thread': 18300, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 27344, 'message': 'yaml file: config\\config.yaml loaded successfully'}sctime)s: INFO: common: yaml file: config\config.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: params.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758795748.9894495, 'msecs': 989.0, 'relativeCreated': 1379.8084259033203, 'thread': 18300, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 27344, 'message': 'yaml file: params.yaml loaded successfully'}sctime)s: INFO: common: yaml file: params.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: schema.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758795748.9913828, 'msecs': 991.0, 'relativeCreated': 1381.7417621612549, 'thread': 18300, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 27344, 'message': 'yaml file: schema.yaml loaded successfully'}sctime)s: INFO: common: yaml file: schema.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758795748.9924214, 'msecs': 992.0, 'relativeCreated': 1382.7803134918213, 'thread': 18300, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 27344, 'message': 'created directory at : artifacts'}sctime)s: INFO: common: created directory at : artifacts]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts/data_ingestion', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758795748.9934216, 'msecs': 993.0, 'relativeCreated': 1383.7804794311523, 'thread': 18300, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 27344, 'message': 'created directory at : artifacts/data_ingestion'}sctime)s: INFO: common: created directory at : artifacts/data_ingestion]
[{'name': 'data_science_project_logger', 'msg': 'FIle already exists', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\components\\data_ingestion.py', 'filename': 'data_ingestion.py', 'module': 'data_ingestion', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 20, 'funcName': 'download_file', 'created': 1758795748.9944239, 'msecs': 994.0, 'relativeCreated': 1384.7827911376953, 'thread': 18300, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 27344, 'message': 'FIle already exists'}sctime)s: INFO: data_ingestion: FIle already exists]
[{'name': 'data_science_project_logger', 'msg': 'extracted all the files to : artifacts/data_ingestion', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\components\\data_ingestion.py', 'filename': 'data_ingestion.py', 'module': 'data_ingestion', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 27, 'funcName': 'extract_zip_file', 'created': 1758795748.998177, 'msecs': 998.0, 'relativeCreated': 1388.535976409912, 'thread': 18300, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 27344, 'message': 'extracted all the files to : artifacts/data_ingestion'}sctime)s: INFO: data_ingestion: extracted all the files to : artifacts/data_ingestion]
[{'name': 'data_science_project_logger', 'msg': '>>>>> stage Data Ingestion Stage completed <<<<<\n\nx==========x', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 19, 'funcName': '<module>', 'created': 1758795748.9987025, 'msecs': 998.0, 'relativeCreated': 1389.061450958252, 'thread': 18300, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 27344, 'message': '>>>>> stage Data Ingestion Stage completed <<<<<\n\nx==========x'}sctime)s: INFO: main: >>>>> stage Data Ingestion Stage completed <<<<<

x==========x]
[{'name': 'data_science_project_logger', 'msg': '>>>>> stage Data Validation Stage started <<<<<', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 26, 'funcName': '<module>', 'created': 1758795748.999227, 'msecs': 999.0, 'relativeCreated': 1389.5859718322754, 'thread': 18300, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 27344, 'message': '>>>>> stage Data Validation Stage started <<<<<'}sctime)s: INFO: main: >>>>> stage Data Validation Stage started <<<<<]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: config\\config.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758795749.0008516, 'msecs': 0.0, 'relativeCreated': 1391.2105560302734, 'thread': 18300, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 27344, 'message': 'yaml file: config\\config.yaml loaded successfully'}sctime)s: INFO: common: yaml file: config\config.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: params.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758795749.0014222, 'msecs': 1.0, 'relativeCreated': 1391.7810916900635, 'thread': 18300, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 27344, 'message': 'yaml file: params.yaml loaded successfully'}sctime)s: INFO: common: yaml file: params.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: schema.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758795749.0036125, 'msecs': 3.0, 'relativeCreated': 1393.9714431762695, 'thread': 18300, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 27344, 'message': 'yaml file: schema.yaml loaded successfully'}sctime)s: INFO: common: yaml file: schema.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758795749.0046585, 'msecs': 4.0, 'relativeCreated': 1395.017385482788, 'thread': 18300, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 27344, 'message': 'created directory at : artifacts'}sctime)s: INFO: common: created directory at : artifacts]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts/data_validation', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758795749.0057702, 'msecs': 5.0, 'relativeCreated': 1396.1291313171387, 'thread': 18300, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 27344, 'message': 'created directory at : artifacts/data_validation'}sctime)s: INFO: common: created directory at : artifacts/data_validation]
[{'name': 'data_science_project_logger', 'msg': 'Data Validation is completed and all the columns are present in the data', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\pipeline\\data_validation_pipeline.py', 'filename': 'data_validation_pipeline.py', 'module': 'data_validation_pipeline', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 20, 'funcName': 'initiate_data_validation', 'created': 1758795749.0321288, 'msecs': 32.0, 'relativeCreated': 1422.487735748291, 'thread': 18300, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 27344, 'message': 'Data Validation is completed and all the columns are present in the data'}sctime)s: INFO: data_validation_pipeline: Data Validation is completed and all the columns are present in the data]
[{'name': 'data_science_project_logger', 'msg': '>>>>> stage Data Validation Stage completed <<<<<\n\nx==========x', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 29, 'funcName': '<module>', 'created': 1758795749.0321288, 'msecs': 32.0, 'relativeCreated': 1422.487735748291, 'thread': 18300, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 27344, 'message': '>>>>> stage Data Validation Stage completed <<<<<\n\nx==========x'}sctime)s: INFO: main: >>>>> stage Data Validation Stage completed <<<<<

x==========x]
[{'name': 'data_science_project_logger', 'msg': '>>>>> stage Data Transformation Stage started <<<<<', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 36, 'funcName': '<module>', 'created': 1758795749.0341413, 'msecs': 34.0, 'relativeCreated': 1424.5002269744873, 'thread': 18300, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 27344, 'message': '>>>>> stage Data Transformation Stage started <<<<<'}sctime)s: INFO: main: >>>>> stage Data Transformation Stage started <<<<<]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: config\\config.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758795749.0355914, 'msecs': 35.0, 'relativeCreated': 1425.950288772583, 'thread': 18300, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 27344, 'message': 'yaml file: config\\config.yaml loaded successfully'}sctime)s: INFO: common: yaml file: config\config.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: params.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758795749.0366378, 'msecs': 36.0, 'relativeCreated': 1426.9967079162598, 'thread': 18300, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 27344, 'message': 'yaml file: params.yaml loaded successfully'}sctime)s: INFO: common: yaml file: params.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: schema.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758795749.0383651, 'msecs': 38.0, 'relativeCreated': 1428.7240505218506, 'thread': 18300, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 27344, 'message': 'yaml file: schema.yaml loaded successfully'}sctime)s: INFO: common: yaml file: schema.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758795749.039409, 'msecs': 39.0, 'relativeCreated': 1429.7678470611572, 'thread': 18300, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 27344, 'message': 'created directory at : artifacts'}sctime)s: INFO: common: created directory at : artifacts]
[{'name': 'data_science_project_logger', 'msg': BoxKeyError("'ConfigBox' object has no attribute 'data_transformation'"), 'args': (), 'levelname': 'ERROR', 'levelno': 40, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': (<class 'box.exceptions.BoxKeyError'>, BoxKeyError("'ConfigBox' object has no attribute 'data_transformation'"), <traceback object at 0x0000022E7F681AC0>), 'exc_text': None, 'stack_info': None, 'lineno': 41, 'funcName': '<module>', 'created': 1758795749.0401452, 'msecs': 40.0, 'relativeCreated': 1430.5040836334229, 'thread': 18300, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 27344, 'message': '"\'ConfigBox\' object has no attribute \'data_transformation\'"'}sctime)s: ERROR: main: "'ConfigBox' object has no attribute 'data_transformation'"]
Traceback (most recent call last):
  File "box\\box.py", line 594, in box.box.Box.__getitem__
KeyError: 'data_transformation'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "box\\box.py", line 633, in box.box.Box.__getattr__
  File "box\\box.py", line 621, in box.box.Box.__getitem__
box.exceptions.BoxKeyError: "'data_transformation'"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "box\\box.py", line 635, in box.box.Box.__getattr__
AttributeError: 'ConfigBox' object has no attribute 'data_transformation'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "box\\config_box.py", line 29, in box.config_box.ConfigBox.__getattr__
  File "box\\box.py", line 649, in box.box.Box.__getattr__
box.exceptions.BoxKeyError: "'ConfigBox' object has no attribute 'data_transformation'"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "box\\box.py", line 594, in box.box.Box.__getitem__
KeyError: 'data_transformation'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "box\\box.py", line 633, in box.box.Box.__getattr__
  File "box\\box.py", line 621, in box.box.Box.__getitem__
box.exceptions.BoxKeyError: "'data_transformation'"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "box\\box.py", line 635, in box.box.Box.__getattr__
AttributeError: 'ConfigBox' object has no attribute 'data_transformation'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "D:\Data Science\main.py", line 38, in <module>
    obj.initiate_data_transformation()
  File "D:\Data Science\src\DataScienceProject\pipeline\data_transformation_pipeline.py", line 21, in initiate_data_transformation
    raise e
  File "D:\Data Science\src\DataScienceProject\pipeline\data_transformation_pipeline.py", line 17, in initiate_data_transformation
    data_transformation_config=config.get_data_transformation_config()
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Data Science\src\DataScienceProject\config\configuration.py", line 41, in get_data_transformation_config
    config=self.config.data_transformation
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "box\\config_box.py", line 31, in box.config_box.ConfigBox.__getattr__
  File "box\\box.py", line 649, in box.box.Box.__getattr__
box.exceptions.BoxKeyError: "'ConfigBox' object has no attribute 'data_transformation'"
[{'name': 'data_science_project_logger', 'msg': '>>>>> stage Data Ingestion Stage started <<<<<', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 16, 'funcName': '<module>', 'created': 1758796001.9716914, 'msecs': 971.0, 'relativeCreated': 1246.6773986816406, 'thread': 21668, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 29968, 'message': '>>>>> stage Data Ingestion Stage started <<<<<'}sctime)s: INFO: main: >>>>> stage Data Ingestion Stage started <<<<<]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: config\\config.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796001.9958198, 'msecs': 995.0, 'relativeCreated': 1270.805835723877, 'thread': 21668, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 29968, 'message': 'yaml file: config\\config.yaml loaded successfully'}sctime)s: INFO: common: yaml file: config\config.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: params.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796001.9958198, 'msecs': 995.0, 'relativeCreated': 1270.805835723877, 'thread': 21668, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 29968, 'message': 'yaml file: params.yaml loaded successfully'}sctime)s: INFO: common: yaml file: params.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: schema.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796001.9958198, 'msecs': 995.0, 'relativeCreated': 1270.805835723877, 'thread': 21668, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 29968, 'message': 'yaml file: schema.yaml loaded successfully'}sctime)s: INFO: common: yaml file: schema.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758796001.9958198, 'msecs': 995.0, 'relativeCreated': 1270.805835723877, 'thread': 21668, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 29968, 'message': 'created directory at : artifacts'}sctime)s: INFO: common: created directory at : artifacts]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts/data_ingestion', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758796002.0048292, 'msecs': 4.0, 'relativeCreated': 1279.8151969909668, 'thread': 21668, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 29968, 'message': 'created directory at : artifacts/data_ingestion'}sctime)s: INFO: common: created directory at : artifacts/data_ingestion]
[{'name': 'data_science_project_logger', 'msg': 'FIle already exists', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\components\\data_ingestion.py', 'filename': 'data_ingestion.py', 'module': 'data_ingestion', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 20, 'funcName': 'download_file', 'created': 1758796002.0058396, 'msecs': 5.0, 'relativeCreated': 1280.8256149291992, 'thread': 21668, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 29968, 'message': 'FIle already exists'}sctime)s: INFO: data_ingestion: FIle already exists]
[{'name': 'data_science_project_logger', 'msg': 'extracted all the files to : artifacts/data_ingestion', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\components\\data_ingestion.py', 'filename': 'data_ingestion.py', 'module': 'data_ingestion', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 27, 'funcName': 'extract_zip_file', 'created': 1758796002.0207436, 'msecs': 20.0, 'relativeCreated': 1295.729637145996, 'thread': 21668, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 29968, 'message': 'extracted all the files to : artifacts/data_ingestion'}sctime)s: INFO: data_ingestion: extracted all the files to : artifacts/data_ingestion]
[{'name': 'data_science_project_logger', 'msg': '>>>>> stage Data Ingestion Stage completed <<<<<\n\nx==========x', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 19, 'funcName': '<module>', 'created': 1758796002.0207436, 'msecs': 20.0, 'relativeCreated': 1295.729637145996, 'thread': 21668, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 29968, 'message': '>>>>> stage Data Ingestion Stage completed <<<<<\n\nx==========x'}sctime)s: INFO: main: >>>>> stage Data Ingestion Stage completed <<<<<

x==========x]
[{'name': 'data_science_project_logger', 'msg': '>>>>> stage Data Validation Stage started <<<<<', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 26, 'funcName': '<module>', 'created': 1758796002.02385, 'msecs': 23.0, 'relativeCreated': 1298.8359928131104, 'thread': 21668, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 29968, 'message': '>>>>> stage Data Validation Stage started <<<<<'}sctime)s: INFO: main: >>>>> stage Data Validation Stage started <<<<<]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: config\\config.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796002.0258415, 'msecs': 25.0, 'relativeCreated': 1300.8275032043457, 'thread': 21668, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 29968, 'message': 'yaml file: config\\config.yaml loaded successfully'}sctime)s: INFO: common: yaml file: config\config.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: params.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796002.0278423, 'msecs': 27.0, 'relativeCreated': 1302.828311920166, 'thread': 21668, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 29968, 'message': 'yaml file: params.yaml loaded successfully'}sctime)s: INFO: common: yaml file: params.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: schema.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796002.0302837, 'msecs': 30.0, 'relativeCreated': 1305.269718170166, 'thread': 21668, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 29968, 'message': 'yaml file: schema.yaml loaded successfully'}sctime)s: INFO: common: yaml file: schema.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758796002.0302837, 'msecs': 30.0, 'relativeCreated': 1305.269718170166, 'thread': 21668, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 29968, 'message': 'created directory at : artifacts'}sctime)s: INFO: common: created directory at : artifacts]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts/data_validation', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758796002.0302837, 'msecs': 30.0, 'relativeCreated': 1305.269718170166, 'thread': 21668, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 29968, 'message': 'created directory at : artifacts/data_validation'}sctime)s: INFO: common: created directory at : artifacts/data_validation]
[{'name': 'data_science_project_logger', 'msg': 'Data Validation is completed and all the columns are present in the data', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\pipeline\\data_validation_pipeline.py', 'filename': 'data_validation_pipeline.py', 'module': 'data_validation_pipeline', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 20, 'funcName': 'initiate_data_validation', 'created': 1758796002.0604086, 'msecs': 60.0, 'relativeCreated': 1335.3946208953857, 'thread': 21668, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 29968, 'message': 'Data Validation is completed and all the columns are present in the data'}sctime)s: INFO: data_validation_pipeline: Data Validation is completed and all the columns are present in the data]
[{'name': 'data_science_project_logger', 'msg': '>>>>> stage Data Validation Stage completed <<<<<\n\nx==========x', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 29, 'funcName': '<module>', 'created': 1758796002.0604086, 'msecs': 60.0, 'relativeCreated': 1335.3946208953857, 'thread': 21668, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 29968, 'message': '>>>>> stage Data Validation Stage completed <<<<<\n\nx==========x'}sctime)s: INFO: main: >>>>> stage Data Validation Stage completed <<<<<

x==========x]
[{'name': 'data_science_project_logger', 'msg': '>>>>> stage Data Transformation Stage started <<<<<', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 36, 'funcName': '<module>', 'created': 1758796002.0604086, 'msecs': 60.0, 'relativeCreated': 1335.3946208953857, 'thread': 21668, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 29968, 'message': '>>>>> stage Data Transformation Stage started <<<<<'}sctime)s: INFO: main: >>>>> stage Data Transformation Stage started <<<<<]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: config\\config.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796002.0604086, 'msecs': 60.0, 'relativeCreated': 1335.3946208953857, 'thread': 21668, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 29968, 'message': 'yaml file: config\\config.yaml loaded successfully'}sctime)s: INFO: common: yaml file: config\config.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: params.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796002.0604086, 'msecs': 60.0, 'relativeCreated': 1335.3946208953857, 'thread': 21668, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 29968, 'message': 'yaml file: params.yaml loaded successfully'}sctime)s: INFO: common: yaml file: params.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: schema.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796002.0604086, 'msecs': 60.0, 'relativeCreated': 1335.3946208953857, 'thread': 21668, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 29968, 'message': 'yaml file: schema.yaml loaded successfully'}sctime)s: INFO: common: yaml file: schema.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758796002.0604086, 'msecs': 60.0, 'relativeCreated': 1335.3946208953857, 'thread': 21668, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 29968, 'message': 'created directory at : artifacts'}sctime)s: INFO: common: created directory at : artifacts]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts/data_transformation', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758796002.0604086, 'msecs': 60.0, 'relativeCreated': 1335.3946208953857, 'thread': 21668, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 29968, 'message': 'created directory at : artifacts/data_transformation'}sctime)s: INFO: common: created directory at : artifacts/data_transformation]
[{'name': 'data_science_project_logger', 'msg': AttributeError("'DataTransformation' object has no attribute 'test_train_splitting'"), 'args': (), 'levelname': 'ERROR', 'levelno': 40, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': (<class 'AttributeError'>, AttributeError("'DataTransformation' object has no attribute 'test_train_splitting'"), <traceback object at 0x0000022BFCEF3BC0>), 'exc_text': None, 'stack_info': None, 'lineno': 41, 'funcName': '<module>', 'created': 1758796002.0604086, 'msecs': 60.0, 'relativeCreated': 1335.3946208953857, 'thread': 21668, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 29968, 'message': "'DataTransformation' object has no attribute 'test_train_splitting'"}sctime)s: ERROR: main: 'DataTransformation' object has no attribute 'test_train_splitting']
Traceback (most recent call last):
  File "D:\Data Science\main.py", line 38, in <module>
    obj.initiate_data_transformation()
  File "D:\Data Science\src\DataScienceProject\pipeline\data_transformation_pipeline.py", line 21, in initiate_data_transformation
    raise e
  File "D:\Data Science\src\DataScienceProject\pipeline\data_transformation_pipeline.py", line 19, in initiate_data_transformation
    data_transformation.test_train_splitting()
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'DataTransformation' object has no attribute 'test_train_splitting'
[{'name': 'data_science_project_logger', 'msg': '>>>>> stage Data Ingestion Stage started <<<<<', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 16, 'funcName': '<module>', 'created': 1758796143.123947, 'msecs': 123.0, 'relativeCreated': 1433.5200786590576, 'thread': 24368, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 4236, 'message': '>>>>> stage Data Ingestion Stage started <<<<<'}sctime)s: INFO: main: >>>>> stage Data Ingestion Stage started <<<<<]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: config\\config.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796143.123947, 'msecs': 123.0, 'relativeCreated': 1433.5200786590576, 'thread': 24368, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 4236, 'message': 'yaml file: config\\config.yaml loaded successfully'}sctime)s: INFO: common: yaml file: config\config.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: params.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796143.1389742, 'msecs': 138.0, 'relativeCreated': 1448.54736328125, 'thread': 24368, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 4236, 'message': 'yaml file: params.yaml loaded successfully'}sctime)s: INFO: common: yaml file: params.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: schema.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796143.1404135, 'msecs': 140.0, 'relativeCreated': 1449.9866962432861, 'thread': 24368, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 4236, 'message': 'yaml file: schema.yaml loaded successfully'}sctime)s: INFO: common: yaml file: schema.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758796143.1404135, 'msecs': 140.0, 'relativeCreated': 1449.9866962432861, 'thread': 24368, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 4236, 'message': 'created directory at : artifacts'}sctime)s: INFO: common: created directory at : artifacts]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts/data_ingestion', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758796143.1404135, 'msecs': 140.0, 'relativeCreated': 1449.9866962432861, 'thread': 24368, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 4236, 'message': 'created directory at : artifacts/data_ingestion'}sctime)s: INFO: common: created directory at : artifacts/data_ingestion]
[{'name': 'data_science_project_logger', 'msg': 'FIle already exists', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\components\\data_ingestion.py', 'filename': 'data_ingestion.py', 'module': 'data_ingestion', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 20, 'funcName': 'download_file', 'created': 1758796143.1404135, 'msecs': 140.0, 'relativeCreated': 1449.9866962432861, 'thread': 24368, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 4236, 'message': 'FIle already exists'}sctime)s: INFO: data_ingestion: FIle already exists]
[{'name': 'data_science_project_logger', 'msg': 'extracted all the files to : artifacts/data_ingestion', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\components\\data_ingestion.py', 'filename': 'data_ingestion.py', 'module': 'data_ingestion', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 27, 'funcName': 'extract_zip_file', 'created': 1758796143.144615, 'msecs': 144.0, 'relativeCreated': 1454.1881084442139, 'thread': 24368, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 4236, 'message': 'extracted all the files to : artifacts/data_ingestion'}sctime)s: INFO: data_ingestion: extracted all the files to : artifacts/data_ingestion]
[{'name': 'data_science_project_logger', 'msg': '>>>>> stage Data Ingestion Stage completed <<<<<\n\nx==========x', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 19, 'funcName': '<module>', 'created': 1758796143.144615, 'msecs': 144.0, 'relativeCreated': 1454.1881084442139, 'thread': 24368, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 4236, 'message': '>>>>> stage Data Ingestion Stage completed <<<<<\n\nx==========x'}sctime)s: INFO: main: >>>>> stage Data Ingestion Stage completed <<<<<

x==========x]
[{'name': 'data_science_project_logger', 'msg': '>>>>> stage Data Validation Stage started <<<<<', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 26, 'funcName': '<module>', 'created': 1758796143.1455817, 'msecs': 145.0, 'relativeCreated': 1455.1548957824707, 'thread': 24368, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 4236, 'message': '>>>>> stage Data Validation Stage started <<<<<'}sctime)s: INFO: main: >>>>> stage Data Validation Stage started <<<<<]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: config\\config.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796143.146598, 'msecs': 146.0, 'relativeCreated': 1456.1712741851807, 'thread': 24368, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 4236, 'message': 'yaml file: config\\config.yaml loaded successfully'}sctime)s: INFO: common: yaml file: config\config.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: params.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796143.146598, 'msecs': 146.0, 'relativeCreated': 1456.1712741851807, 'thread': 24368, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 4236, 'message': 'yaml file: params.yaml loaded successfully'}sctime)s: INFO: common: yaml file: params.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: schema.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796143.1486464, 'msecs': 148.0, 'relativeCreated': 1458.2195281982422, 'thread': 24368, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 4236, 'message': 'yaml file: schema.yaml loaded successfully'}sctime)s: INFO: common: yaml file: schema.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758796143.1486464, 'msecs': 148.0, 'relativeCreated': 1458.2195281982422, 'thread': 24368, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 4236, 'message': 'created directory at : artifacts'}sctime)s: INFO: common: created directory at : artifacts]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts/data_validation', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758796143.1486464, 'msecs': 148.0, 'relativeCreated': 1458.2195281982422, 'thread': 24368, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 4236, 'message': 'created directory at : artifacts/data_validation'}sctime)s: INFO: common: created directory at : artifacts/data_validation]
[{'name': 'data_science_project_logger', 'msg': 'Data Validation is completed and all the columns are present in the data', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\pipeline\\data_validation_pipeline.py', 'filename': 'data_validation_pipeline.py', 'module': 'data_validation_pipeline', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 20, 'funcName': 'initiate_data_validation', 'created': 1758796143.17809, 'msecs': 178.0, 'relativeCreated': 1487.6632690429688, 'thread': 24368, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 4236, 'message': 'Data Validation is completed and all the columns are present in the data'}sctime)s: INFO: data_validation_pipeline: Data Validation is completed and all the columns are present in the data]
[{'name': 'data_science_project_logger', 'msg': '>>>>> stage Data Validation Stage completed <<<<<\n\nx==========x', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 29, 'funcName': '<module>', 'created': 1758796143.17809, 'msecs': 178.0, 'relativeCreated': 1487.6632690429688, 'thread': 24368, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 4236, 'message': '>>>>> stage Data Validation Stage completed <<<<<\n\nx==========x'}sctime)s: INFO: main: >>>>> stage Data Validation Stage completed <<<<<

x==========x]
[{'name': 'data_science_project_logger', 'msg': '>>>>> stage Data Transformation Stage started <<<<<', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 36, 'funcName': '<module>', 'created': 1758796143.17809, 'msecs': 178.0, 'relativeCreated': 1487.6632690429688, 'thread': 24368, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 4236, 'message': '>>>>> stage Data Transformation Stage started <<<<<'}sctime)s: INFO: main: >>>>> stage Data Transformation Stage started <<<<<]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: config\\config.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796143.17809, 'msecs': 178.0, 'relativeCreated': 1487.6632690429688, 'thread': 24368, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 4236, 'message': 'yaml file: config\\config.yaml loaded successfully'}sctime)s: INFO: common: yaml file: config\config.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: params.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796143.17809, 'msecs': 178.0, 'relativeCreated': 1487.6632690429688, 'thread': 24368, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 4236, 'message': 'yaml file: params.yaml loaded successfully'}sctime)s: INFO: common: yaml file: params.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: schema.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796143.17809, 'msecs': 178.0, 'relativeCreated': 1487.6632690429688, 'thread': 24368, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 4236, 'message': 'yaml file: schema.yaml loaded successfully'}sctime)s: INFO: common: yaml file: schema.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758796143.17809, 'msecs': 178.0, 'relativeCreated': 1487.6632690429688, 'thread': 24368, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 4236, 'message': 'created directory at : artifacts'}sctime)s: INFO: common: created directory at : artifacts]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts/data_transformation', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758796143.1844761, 'msecs': 184.0, 'relativeCreated': 1494.049310684204, 'thread': 24368, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 4236, 'message': 'created directory at : artifacts/data_transformation'}sctime)s: INFO: common: created directory at : artifacts/data_transformation]
[{'name': 'data_science_project_logger', 'msg': AttributeError("'DataTransformationConfig' object has no attribute 'root_dir_path'"), 'args': (), 'levelname': 'ERROR', 'levelno': 40, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': (<class 'AttributeError'>, AttributeError("'DataTransformationConfig' object has no attribute 'root_dir_path'"), <traceback object at 0x000001F9FFD3EC80>), 'exc_text': None, 'stack_info': None, 'lineno': 41, 'funcName': '<module>', 'created': 1758796143.1983936, 'msecs': 198.0, 'relativeCreated': 1507.9667568206787, 'thread': 24368, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 4236, 'message': "'DataTransformationConfig' object has no attribute 'root_dir_path'"}sctime)s: ERROR: main: 'DataTransformationConfig' object has no attribute 'root_dir_path']
Traceback (most recent call last):
  File "D:\Data Science\main.py", line 38, in <module>
    obj.initiate_data_transformation()
  File "D:\Data Science\src\DataScienceProject\pipeline\data_transformation_pipeline.py", line 21, in initiate_data_transformation
    raise e
  File "D:\Data Science\src\DataScienceProject\pipeline\data_transformation_pipeline.py", line 19, in initiate_data_transformation
    data_transformation.train_test_splitting()
  File "D:\Data Science\src\DataScienceProject\components\data_transformation.py", line 14, in train_test_splitting
    train.to_csv(self.config.root_dir_path,index=False)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'DataTransformationConfig' object has no attribute 'root_dir_path'
[{'name': 'data_science_project_logger', 'msg': '>>>>> stage Data Ingestion Stage started <<<<<', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 16, 'funcName': '<module>', 'created': 1758796234.3901923, 'msecs': 390.0, 'relativeCreated': 1365.3781414031982, 'thread': 30344, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 21496, 'message': '>>>>> stage Data Ingestion Stage started <<<<<'}sctime)s: INFO: main: >>>>> stage Data Ingestion Stage started <<<<<]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: config\\config.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796234.3924928, 'msecs': 392.0, 'relativeCreated': 1367.6786422729492, 'thread': 30344, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 21496, 'message': 'yaml file: config\\config.yaml loaded successfully'}sctime)s: INFO: common: yaml file: config\config.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: params.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796234.3924928, 'msecs': 392.0, 'relativeCreated': 1367.6786422729492, 'thread': 30344, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 21496, 'message': 'yaml file: params.yaml loaded successfully'}sctime)s: INFO: common: yaml file: params.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: schema.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796234.3934913, 'msecs': 393.0, 'relativeCreated': 1368.6771392822266, 'thread': 30344, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 21496, 'message': 'yaml file: schema.yaml loaded successfully'}sctime)s: INFO: common: yaml file: schema.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758796234.3944926, 'msecs': 394.0, 'relativeCreated': 1369.6784973144531, 'thread': 30344, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 21496, 'message': 'created directory at : artifacts'}sctime)s: INFO: common: created directory at : artifacts]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts/data_ingestion', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758796234.3944926, 'msecs': 394.0, 'relativeCreated': 1369.6784973144531, 'thread': 30344, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 21496, 'message': 'created directory at : artifacts/data_ingestion'}sctime)s: INFO: common: created directory at : artifacts/data_ingestion]
[{'name': 'data_science_project_logger', 'msg': 'FIle already exists', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\components\\data_ingestion.py', 'filename': 'data_ingestion.py', 'module': 'data_ingestion', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 20, 'funcName': 'download_file', 'created': 1758796234.3954926, 'msecs': 395.0, 'relativeCreated': 1370.678424835205, 'thread': 30344, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 21496, 'message': 'FIle already exists'}sctime)s: INFO: data_ingestion: FIle already exists]
[{'name': 'data_science_project_logger', 'msg': 'extracted all the files to : artifacts/data_ingestion', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\components\\data_ingestion.py', 'filename': 'data_ingestion.py', 'module': 'data_ingestion', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 27, 'funcName': 'extract_zip_file', 'created': 1758796234.3974934, 'msecs': 397.0, 'relativeCreated': 1372.6792335510254, 'thread': 30344, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 21496, 'message': 'extracted all the files to : artifacts/data_ingestion'}sctime)s: INFO: data_ingestion: extracted all the files to : artifacts/data_ingestion]
[{'name': 'data_science_project_logger', 'msg': '>>>>> stage Data Ingestion Stage completed <<<<<\n\nx==========x', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 19, 'funcName': '<module>', 'created': 1758796234.3974934, 'msecs': 397.0, 'relativeCreated': 1372.6792335510254, 'thread': 30344, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 21496, 'message': '>>>>> stage Data Ingestion Stage completed <<<<<\n\nx==========x'}sctime)s: INFO: main: >>>>> stage Data Ingestion Stage completed <<<<<

x==========x]
[{'name': 'data_science_project_logger', 'msg': '>>>>> stage Data Validation Stage started <<<<<', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 26, 'funcName': '<module>', 'created': 1758796234.3974934, 'msecs': 397.0, 'relativeCreated': 1372.6792335510254, 'thread': 30344, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 21496, 'message': '>>>>> stage Data Validation Stage started <<<<<'}sctime)s: INFO: main: >>>>> stage Data Validation Stage started <<<<<]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: config\\config.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796234.399493, 'msecs': 399.0, 'relativeCreated': 1374.6788501739502, 'thread': 30344, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 21496, 'message': 'yaml file: config\\config.yaml loaded successfully'}sctime)s: INFO: common: yaml file: config\config.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: params.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796234.399493, 'msecs': 399.0, 'relativeCreated': 1374.6788501739502, 'thread': 30344, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 21496, 'message': 'yaml file: params.yaml loaded successfully'}sctime)s: INFO: common: yaml file: params.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: schema.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796234.401492, 'msecs': 401.0, 'relativeCreated': 1376.6779899597168, 'thread': 30344, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 21496, 'message': 'yaml file: schema.yaml loaded successfully'}sctime)s: INFO: common: yaml file: schema.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758796234.401492, 'msecs': 401.0, 'relativeCreated': 1376.6779899597168, 'thread': 30344, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 21496, 'message': 'created directory at : artifacts'}sctime)s: INFO: common: created directory at : artifacts]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts/data_validation', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758796234.4024918, 'msecs': 402.0, 'relativeCreated': 1377.6776790618896, 'thread': 30344, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 21496, 'message': 'created directory at : artifacts/data_validation'}sctime)s: INFO: common: created directory at : artifacts/data_validation]
[{'name': 'data_science_project_logger', 'msg': 'Data Validation is completed and all the columns are present in the data', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\pipeline\\data_validation_pipeline.py', 'filename': 'data_validation_pipeline.py', 'module': 'data_validation_pipeline', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 20, 'funcName': 'initiate_data_validation', 'created': 1758796234.4348822, 'msecs': 434.0, 'relativeCreated': 1410.0680351257324, 'thread': 30344, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 21496, 'message': 'Data Validation is completed and all the columns are present in the data'}sctime)s: INFO: data_validation_pipeline: Data Validation is completed and all the columns are present in the data]
[{'name': 'data_science_project_logger', 'msg': '>>>>> stage Data Validation Stage completed <<<<<\n\nx==========x', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 29, 'funcName': '<module>', 'created': 1758796234.4348822, 'msecs': 434.0, 'relativeCreated': 1410.0680351257324, 'thread': 30344, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 21496, 'message': '>>>>> stage Data Validation Stage completed <<<<<\n\nx==========x'}sctime)s: INFO: main: >>>>> stage Data Validation Stage completed <<<<<

x==========x]
[{'name': 'data_science_project_logger', 'msg': '>>>>> stage Data Transformation Stage started <<<<<', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 36, 'funcName': '<module>', 'created': 1758796234.4348822, 'msecs': 434.0, 'relativeCreated': 1410.0680351257324, 'thread': 30344, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 21496, 'message': '>>>>> stage Data Transformation Stage started <<<<<'}sctime)s: INFO: main: >>>>> stage Data Transformation Stage started <<<<<]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: config\\config.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796234.4348822, 'msecs': 434.0, 'relativeCreated': 1410.0680351257324, 'thread': 30344, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 21496, 'message': 'yaml file: config\\config.yaml loaded successfully'}sctime)s: INFO: common: yaml file: config\config.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: params.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796234.4348822, 'msecs': 434.0, 'relativeCreated': 1410.0680351257324, 'thread': 30344, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 21496, 'message': 'yaml file: params.yaml loaded successfully'}sctime)s: INFO: common: yaml file: params.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: schema.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796234.4394648, 'msecs': 439.0, 'relativeCreated': 1414.6506786346436, 'thread': 30344, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 21496, 'message': 'yaml file: schema.yaml loaded successfully'}sctime)s: INFO: common: yaml file: schema.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758796234.4412751, 'msecs': 441.0, 'relativeCreated': 1416.4609909057617, 'thread': 30344, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 21496, 'message': 'created directory at : artifacts'}sctime)s: INFO: common: created directory at : artifacts]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts/data_transformation', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758796234.4412751, 'msecs': 441.0, 'relativeCreated': 1416.4609909057617, 'thread': 30344, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 21496, 'message': 'created directory at : artifacts/data_transformation'}sctime)s: INFO: common: created directory at : artifacts/data_transformation]
[{'name': 'data_science_project_logger', 'msg': PermissionError(13, 'Permission denied'), 'args': (), 'levelname': 'ERROR', 'levelno': 40, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': (<class 'PermissionError'>, PermissionError(13, 'Permission denied'), <traceback object at 0x000001B382860540>), 'exc_text': None, 'stack_info': None, 'lineno': 41, 'funcName': '<module>', 'created': 1758796234.470438, 'msecs': 470.0, 'relativeCreated': 1445.6238746643066, 'thread': 30344, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 21496, 'message': "[Errno 13] Permission denied: 'artifacts/data_transformation'"}sctime)s: ERROR: main: [Errno 13] Permission denied: 'artifacts/data_transformation']
Traceback (most recent call last):
  File "D:\Data Science\main.py", line 38, in <module>
    obj.initiate_data_transformation()
  File "D:\Data Science\src\DataScienceProject\pipeline\data_transformation_pipeline.py", line 21, in initiate_data_transformation
    raise e
  File "D:\Data Science\src\DataScienceProject\pipeline\data_transformation_pipeline.py", line 19, in initiate_data_transformation
    data_transformation.train_test_splitting()
  File "D:\Data Science\src\DataScienceProject\components\data_transformation.py", line 14, in train_test_splitting
    train.to_csv(self.config.root_dir,index=False)
  File "D:\Data Science\.venv\Lib\site-packages\pandas\util\_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "D:\Data Science\.venv\Lib\site-packages\pandas\core\generic.py", line 3986, in to_csv
    return DataFrameRenderer(formatter).to_csv(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Data Science\.venv\Lib\site-packages\pandas\io\formats\format.py", line 1014, in to_csv
    csv_formatter.save()
  File "D:\Data Science\.venv\Lib\site-packages\pandas\io\formats\csvs.py", line 251, in save
    with get_handle(
         ^^^^^^^^^^^
  File "D:\Data Science\.venv\Lib\site-packages\pandas\io\common.py", line 873, in get_handle
    handle = open(
             ^^^^^
PermissionError: [Errno 13] Permission denied: 'artifacts/data_transformation'
[{'name': 'data_science_project_logger', 'msg': '>>>>> stage Data Ingestion Stage started <<<<<', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 16, 'funcName': '<module>', 'created': 1758796251.5331278, 'msecs': 533.0, 'relativeCreated': 1320.957899093628, 'thread': 8524, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 26196, 'message': '>>>>> stage Data Ingestion Stage started <<<<<'}sctime)s: INFO: main: >>>>> stage Data Ingestion Stage started <<<<<]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: config\\config.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796251.5341132, 'msecs': 534.0, 'relativeCreated': 1321.9432830810547, 'thread': 8524, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 26196, 'message': 'yaml file: config\\config.yaml loaded successfully'}sctime)s: INFO: common: yaml file: config\config.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: params.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796251.534848, 'msecs': 534.0, 'relativeCreated': 1322.6780891418457, 'thread': 8524, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 26196, 'message': 'yaml file: params.yaml loaded successfully'}sctime)s: INFO: common: yaml file: params.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: schema.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796251.5358646, 'msecs': 535.0, 'relativeCreated': 1323.6947059631348, 'thread': 8524, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 26196, 'message': 'yaml file: schema.yaml loaded successfully'}sctime)s: INFO: common: yaml file: schema.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758796251.5366335, 'msecs': 536.0, 'relativeCreated': 1324.4636058807373, 'thread': 8524, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 26196, 'message': 'created directory at : artifacts'}sctime)s: INFO: common: created directory at : artifacts]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts/data_ingestion', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758796251.5366335, 'msecs': 536.0, 'relativeCreated': 1324.4636058807373, 'thread': 8524, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 26196, 'message': 'created directory at : artifacts/data_ingestion'}sctime)s: INFO: common: created directory at : artifacts/data_ingestion]
[{'name': 'data_science_project_logger', 'msg': 'FIle already exists', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\components\\data_ingestion.py', 'filename': 'data_ingestion.py', 'module': 'data_ingestion', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 20, 'funcName': 'download_file', 'created': 1758796251.5376499, 'msecs': 537.0, 'relativeCreated': 1325.4799842834473, 'thread': 8524, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 26196, 'message': 'FIle already exists'}sctime)s: INFO: data_ingestion: FIle already exists]
[{'name': 'data_science_project_logger', 'msg': 'extracted all the files to : artifacts/data_ingestion', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\components\\data_ingestion.py', 'filename': 'data_ingestion.py', 'module': 'data_ingestion', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 27, 'funcName': 'extract_zip_file', 'created': 1758796251.5403569, 'msecs': 540.0, 'relativeCreated': 1328.1869888305664, 'thread': 8524, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 26196, 'message': 'extracted all the files to : artifacts/data_ingestion'}sctime)s: INFO: data_ingestion: extracted all the files to : artifacts/data_ingestion]
[{'name': 'data_science_project_logger', 'msg': '>>>>> stage Data Ingestion Stage completed <<<<<\n\nx==========x', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 19, 'funcName': '<module>', 'created': 1758796251.5433717, 'msecs': 543.0, 'relativeCreated': 1331.2017917633057, 'thread': 8524, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 26196, 'message': '>>>>> stage Data Ingestion Stage completed <<<<<\n\nx==========x'}sctime)s: INFO: main: >>>>> stage Data Ingestion Stage completed <<<<<

x==========x]
[{'name': 'data_science_project_logger', 'msg': '>>>>> stage Data Validation Stage started <<<<<', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 26, 'funcName': '<module>', 'created': 1758796251.5433717, 'msecs': 543.0, 'relativeCreated': 1331.2017917633057, 'thread': 8524, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 26196, 'message': '>>>>> stage Data Validation Stage started <<<<<'}sctime)s: INFO: main: >>>>> stage Data Validation Stage started <<<<<]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: config\\config.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796251.5433717, 'msecs': 543.0, 'relativeCreated': 1331.2017917633057, 'thread': 8524, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 26196, 'message': 'yaml file: config\\config.yaml loaded successfully'}sctime)s: INFO: common: yaml file: config\config.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: params.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796251.546863, 'msecs': 546.0, 'relativeCreated': 1334.693193435669, 'thread': 8524, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 26196, 'message': 'yaml file: params.yaml loaded successfully'}sctime)s: INFO: common: yaml file: params.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: schema.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796251.546863, 'msecs': 546.0, 'relativeCreated': 1334.693193435669, 'thread': 8524, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 26196, 'message': 'yaml file: schema.yaml loaded successfully'}sctime)s: INFO: common: yaml file: schema.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758796251.546863, 'msecs': 546.0, 'relativeCreated': 1334.693193435669, 'thread': 8524, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 26196, 'message': 'created directory at : artifacts'}sctime)s: INFO: common: created directory at : artifacts]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts/data_validation', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758796251.546863, 'msecs': 546.0, 'relativeCreated': 1334.693193435669, 'thread': 8524, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 26196, 'message': 'created directory at : artifacts/data_validation'}sctime)s: INFO: common: created directory at : artifacts/data_validation]
[{'name': 'data_science_project_logger', 'msg': 'Data Validation is completed and all the columns are present in the data', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\pipeline\\data_validation_pipeline.py', 'filename': 'data_validation_pipeline.py', 'module': 'data_validation_pipeline', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 20, 'funcName': 'initiate_data_validation', 'created': 1758796251.5773003, 'msecs': 577.0, 'relativeCreated': 1365.1304244995117, 'thread': 8524, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 26196, 'message': 'Data Validation is completed and all the columns are present in the data'}sctime)s: INFO: data_validation_pipeline: Data Validation is completed and all the columns are present in the data]
[{'name': 'data_science_project_logger', 'msg': '>>>>> stage Data Validation Stage completed <<<<<\n\nx==========x', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 29, 'funcName': '<module>', 'created': 1758796251.5773003, 'msecs': 577.0, 'relativeCreated': 1365.1304244995117, 'thread': 8524, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 26196, 'message': '>>>>> stage Data Validation Stage completed <<<<<\n\nx==========x'}sctime)s: INFO: main: >>>>> stage Data Validation Stage completed <<<<<

x==========x]
[{'name': 'data_science_project_logger', 'msg': '>>>>> stage Data Transformation Stage started <<<<<', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 36, 'funcName': '<module>', 'created': 1758796251.5773003, 'msecs': 577.0, 'relativeCreated': 1365.1304244995117, 'thread': 8524, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 26196, 'message': '>>>>> stage Data Transformation Stage started <<<<<'}sctime)s: INFO: main: >>>>> stage Data Transformation Stage started <<<<<]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: config\\config.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796251.5773003, 'msecs': 577.0, 'relativeCreated': 1365.1304244995117, 'thread': 8524, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 26196, 'message': 'yaml file: config\\config.yaml loaded successfully'}sctime)s: INFO: common: yaml file: config\config.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: params.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796251.5773003, 'msecs': 577.0, 'relativeCreated': 1365.1304244995117, 'thread': 8524, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 26196, 'message': 'yaml file: params.yaml loaded successfully'}sctime)s: INFO: common: yaml file: params.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: schema.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796251.5773003, 'msecs': 577.0, 'relativeCreated': 1365.1304244995117, 'thread': 8524, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 26196, 'message': 'yaml file: schema.yaml loaded successfully'}sctime)s: INFO: common: yaml file: schema.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758796251.5773003, 'msecs': 577.0, 'relativeCreated': 1365.1304244995117, 'thread': 8524, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 26196, 'message': 'created directory at : artifacts'}sctime)s: INFO: common: created directory at : artifacts]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts/data_transformation', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758796251.5773003, 'msecs': 577.0, 'relativeCreated': 1365.1304244995117, 'thread': 8524, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 26196, 'message': 'created directory at : artifacts/data_transformation'}sctime)s: INFO: common: created directory at : artifacts/data_transformation]
[{'name': 'data_science_project_logger', 'msg': PermissionError(13, 'Permission denied'), 'args': (), 'levelname': 'ERROR', 'levelno': 40, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': (<class 'PermissionError'>, PermissionError(13, 'Permission denied'), <traceback object at 0x00000258A3DF2700>), 'exc_text': None, 'stack_info': None, 'lineno': 41, 'funcName': '<module>', 'created': 1758796251.5926573, 'msecs': 592.0, 'relativeCreated': 1380.4874420166016, 'thread': 8524, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 26196, 'message': "[Errno 13] Permission denied: 'artifacts/data_transformation'"}sctime)s: ERROR: main: [Errno 13] Permission denied: 'artifacts/data_transformation']
Traceback (most recent call last):
  File "D:\Data Science\main.py", line 38, in <module>
    obj.initiate_data_transformation()
  File "D:\Data Science\src\DataScienceProject\pipeline\data_transformation_pipeline.py", line 21, in initiate_data_transformation
    raise e
  File "D:\Data Science\src\DataScienceProject\pipeline\data_transformation_pipeline.py", line 19, in initiate_data_transformation
    data_transformation.train_test_splitting()
  File "D:\Data Science\src\DataScienceProject\components\data_transformation.py", line 14, in train_test_splitting
    train.to_csv(self.config.root_dir,index=False)
  File "D:\Data Science\.venv\Lib\site-packages\pandas\util\_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "D:\Data Science\.venv\Lib\site-packages\pandas\core\generic.py", line 3986, in to_csv
    return DataFrameRenderer(formatter).to_csv(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Data Science\.venv\Lib\site-packages\pandas\io\formats\format.py", line 1014, in to_csv
    csv_formatter.save()
  File "D:\Data Science\.venv\Lib\site-packages\pandas\io\formats\csvs.py", line 251, in save
    with get_handle(
         ^^^^^^^^^^^
  File "D:\Data Science\.venv\Lib\site-packages\pandas\io\common.py", line 873, in get_handle
    handle = open(
             ^^^^^
PermissionError: [Errno 13] Permission denied: 'artifacts/data_transformation'
[{'name': 'data_science_project_logger', 'msg': '>>>>> stage Data Ingestion Stage started <<<<<', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 16, 'funcName': '<module>', 'created': 1758796361.7362812, 'msecs': 736.0, 'relativeCreated': 1261.7056369781494, 'thread': 28460, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 28776, 'message': '>>>>> stage Data Ingestion Stage started <<<<<'}sctime)s: INFO: main: >>>>> stage Data Ingestion Stage started <<<<<]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: config\\config.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796361.73829, 'msecs': 738.0, 'relativeCreated': 1263.7145519256592, 'thread': 28460, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 28776, 'message': 'yaml file: config\\config.yaml loaded successfully'}sctime)s: INFO: common: yaml file: config\config.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: params.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796361.7392714, 'msecs': 739.0, 'relativeCreated': 1264.6958827972412, 'thread': 28460, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 28776, 'message': 'yaml file: params.yaml loaded successfully'}sctime)s: INFO: common: yaml file: params.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: schema.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796361.741435, 'msecs': 741.0, 'relativeCreated': 1266.859531402588, 'thread': 28460, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 28776, 'message': 'yaml file: schema.yaml loaded successfully'}sctime)s: INFO: common: yaml file: schema.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758796361.7424564, 'msecs': 742.0, 'relativeCreated': 1267.880916595459, 'thread': 28460, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 28776, 'message': 'created directory at : artifacts'}sctime)s: INFO: common: created directory at : artifacts]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts/data_ingestion', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758796361.7434437, 'msecs': 743.0, 'relativeCreated': 1268.8682079315186, 'thread': 28460, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 28776, 'message': 'created directory at : artifacts/data_ingestion'}sctime)s: INFO: common: created directory at : artifacts/data_ingestion]
[{'name': 'data_science_project_logger', 'msg': 'FIle already exists', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\components\\data_ingestion.py', 'filename': 'data_ingestion.py', 'module': 'data_ingestion', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 20, 'funcName': 'download_file', 'created': 1758796361.7444506, 'msecs': 744.0, 'relativeCreated': 1269.8750495910645, 'thread': 28460, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 28776, 'message': 'FIle already exists'}sctime)s: INFO: data_ingestion: FIle already exists]
[{'name': 'data_science_project_logger', 'msg': 'extracted all the files to : artifacts/data_ingestion', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\components\\data_ingestion.py', 'filename': 'data_ingestion.py', 'module': 'data_ingestion', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 27, 'funcName': 'extract_zip_file', 'created': 1758796361.7464492, 'msecs': 746.0, 'relativeCreated': 1271.8737125396729, 'thread': 28460, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 28776, 'message': 'extracted all the files to : artifacts/data_ingestion'}sctime)s: INFO: data_ingestion: extracted all the files to : artifacts/data_ingestion]
[{'name': 'data_science_project_logger', 'msg': '>>>>> stage Data Ingestion Stage completed <<<<<\n\nx==========x', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 19, 'funcName': '<module>', 'created': 1758796361.747453, 'msecs': 747.0, 'relativeCreated': 1272.8774547576904, 'thread': 28460, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 28776, 'message': '>>>>> stage Data Ingestion Stage completed <<<<<\n\nx==========x'}sctime)s: INFO: main: >>>>> stage Data Ingestion Stage completed <<<<<

x==========x]
[{'name': 'data_science_project_logger', 'msg': '>>>>> stage Data Validation Stage started <<<<<', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 26, 'funcName': '<module>', 'created': 1758796361.747453, 'msecs': 747.0, 'relativeCreated': 1272.8774547576904, 'thread': 28460, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 28776, 'message': '>>>>> stage Data Validation Stage started <<<<<'}sctime)s: INFO: main: >>>>> stage Data Validation Stage started <<<<<]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: config\\config.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796361.748456, 'msecs': 748.0, 'relativeCreated': 1273.8804817199707, 'thread': 28460, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 28776, 'message': 'yaml file: config\\config.yaml loaded successfully'}sctime)s: INFO: common: yaml file: config\config.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: params.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796361.7494426, 'msecs': 749.0, 'relativeCreated': 1274.867057800293, 'thread': 28460, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 28776, 'message': 'yaml file: params.yaml loaded successfully'}sctime)s: INFO: common: yaml file: params.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: schema.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796361.7524564, 'msecs': 752.0, 'relativeCreated': 1277.8809070587158, 'thread': 28460, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 28776, 'message': 'yaml file: schema.yaml loaded successfully'}sctime)s: INFO: common: yaml file: schema.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758796361.7524564, 'msecs': 752.0, 'relativeCreated': 1277.8809070587158, 'thread': 28460, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 28776, 'message': 'created directory at : artifacts'}sctime)s: INFO: common: created directory at : artifacts]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts/data_validation', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758796361.7534442, 'msecs': 753.0, 'relativeCreated': 1278.8686752319336, 'thread': 28460, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 28776, 'message': 'created directory at : artifacts/data_validation'}sctime)s: INFO: common: created directory at : artifacts/data_validation]
[{'name': 'data_science_project_logger', 'msg': 'Data Validation is completed and all the columns are present in the data', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\pipeline\\data_validation_pipeline.py', 'filename': 'data_validation_pipeline.py', 'module': 'data_validation_pipeline', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 20, 'funcName': 'initiate_data_validation', 'created': 1758796361.7844398, 'msecs': 784.0, 'relativeCreated': 1309.8642826080322, 'thread': 28460, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 28776, 'message': 'Data Validation is completed and all the columns are present in the data'}sctime)s: INFO: data_validation_pipeline: Data Validation is completed and all the columns are present in the data]
[{'name': 'data_science_project_logger', 'msg': '>>>>> stage Data Validation Stage completed <<<<<\n\nx==========x', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 29, 'funcName': '<module>', 'created': 1758796361.7854755, 'msecs': 785.0, 'relativeCreated': 1310.8999729156494, 'thread': 28460, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 28776, 'message': '>>>>> stage Data Validation Stage completed <<<<<\n\nx==========x'}sctime)s: INFO: main: >>>>> stage Data Validation Stage completed <<<<<

x==========x]
[{'name': 'data_science_project_logger', 'msg': '>>>>> stage Data Transformation Stage started <<<<<', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 36, 'funcName': '<module>', 'created': 1758796361.7854755, 'msecs': 785.0, 'relativeCreated': 1310.8999729156494, 'thread': 28460, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 28776, 'message': '>>>>> stage Data Transformation Stage started <<<<<'}sctime)s: INFO: main: >>>>> stage Data Transformation Stage started <<<<<]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: config\\config.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796361.7875009, 'msecs': 787.0, 'relativeCreated': 1312.9253387451172, 'thread': 28460, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 28776, 'message': 'yaml file: config\\config.yaml loaded successfully'}sctime)s: INFO: common: yaml file: config\config.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: params.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796361.7875009, 'msecs': 787.0, 'relativeCreated': 1312.9253387451172, 'thread': 28460, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 28776, 'message': 'yaml file: params.yaml loaded successfully'}sctime)s: INFO: common: yaml file: params.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'yaml file: schema.yaml loaded successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'read_yaml', 'created': 1758796361.7894344, 'msecs': 789.0, 'relativeCreated': 1314.8589134216309, 'thread': 28460, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 28776, 'message': 'yaml file: schema.yaml loaded successfully'}sctime)s: INFO: common: yaml file: schema.yaml loaded successfully]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758796361.7905772, 'msecs': 790.0, 'relativeCreated': 1316.0016536712646, 'thread': 28460, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 28776, 'message': 'created directory at : artifacts'}sctime)s: INFO: common: created directory at : artifacts]
[{'name': 'data_science_project_logger', 'msg': 'created directory at : artifacts/data_transformation', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\utils\\common.py', 'filename': 'common.py', 'module': 'common', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 30, 'funcName': 'create_directories', 'created': 1758796361.792595, 'msecs': 792.0, 'relativeCreated': 1318.0193901062012, 'thread': 28460, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 28776, 'message': 'created directory at : artifacts/data_transformation'}sctime)s: INFO: common: created directory at : artifacts/data_transformation]
[{'name': 'data_science_project_logger', 'msg': 'Split the data into train and test successfully', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\components\\data_transformation.py', 'filename': 'data_transformation.py', 'module': 'data_transformation', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 17, 'funcName': 'train_test_splitting', 'created': 1758796361.8132825, 'msecs': 813.0, 'relativeCreated': 1338.7069702148438, 'thread': 28460, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 28776, 'message': 'Split the data into train and test successfully'}sctime)s: INFO: data_transformation: Split the data into train and test successfully]
[{'name': 'data_science_project_logger', 'msg': 'train data : (1279, 12) and test data : (320, 12)', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\src\\DataScienceProject\\components\\data_transformation.py', 'filename': 'data_transformation.py', 'module': 'data_transformation', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 18, 'funcName': 'train_test_splitting', 'created': 1758796361.813796, 'msecs': 813.0, 'relativeCreated': 1339.2205238342285, 'thread': 28460, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 28776, 'message': 'train data : (1279, 12) and test data : (320, 12)'}sctime)s: INFO: data_transformation: train data : (1279, 12) and test data : (320, 12)]
[{'name': 'data_science_project_logger', 'msg': '>>>>> stage Data Transformation Stage completed <<<<<\n\nx==========x', 'args': (), 'levelname': 'INFO', 'levelno': 20, 'pathname': 'D:\\Data Science\\main.py', 'filename': 'main.py', 'module': 'main', 'exc_info': None, 'exc_text': None, 'stack_info': None, 'lineno': 39, 'funcName': '<module>', 'created': 1758796361.813796, 'msecs': 813.0, 'relativeCreated': 1339.2205238342285, 'thread': 28460, 'threadName': 'MainThread', 'processName': 'MainProcess', 'process': 28776, 'message': '>>>>> stage Data Transformation Stage completed <<<<<\n\nx==========x'}sctime)s: INFO: main: >>>>> stage Data Transformation Stage completed <<<<<

x==========x]
